### DISCUSSION TITLE
looking for a github action

### DISCUSSION THREAD

**Discussion by Samyu09** [open-telemetry/community#1997] [Community Discussions]
i have a github action. **kittychiu/workflow-metrics@v0.4.7**  that generates the metrics of the workflows running in repository 
now i want to send these metrics to Open telemetry any idea on how can i t be done? or do we have any github action already present to do so

**Comment by svrnm** [comment]
Hey @Samyu09, 

two answers to your question:

* There is a SIG that is currently looking into CI/CD + OpenTelemetry, you can find a project description and additional resources here: https://github.com/open-telemetry/community/blob/main/projects/ci-cd.md (there's also a link to an issue for an opentelemetry collector for github actions)
* Since the github action you mentioned is written in Python, you _might_(!) be able to add the OTel Python SDK to it to emit telemetry to an otel collector/other backend

**Reply by Samyu09** [reply]
Hey @svrnm 

1. i would like to explain my workflow so you will may be understand what i am actually looking for.
2. I need some github action that generates **workflow metrics** not traces , i want these metrics to be pushed to open telemetry so i can send it to prometheus at my backend.
3. The one you provided is all relating traces.
4. May be you can skip the python action that i pointed earlier.

**Reply by svrnm** [reply]
Hey @Samyu09, feel free and go ahead to explain your workflow. The documents I pointed you to are what we have as of today, and what we are working on. You can join this community or comment on the appropriate issues and help with the metrics part.

**Reply by Samyu09** [reply]
@svrnm 
so you mean there's no such github action till date to do so????
myflow goes like this  **github actions -> opentelemetry collector -> prometheus**
but i wonder if theres some github action created already so as to send the workflow metrics. If no please let me know other ways to do so

**Reply by Samyu09** [reply]
As part of my work to analyze the workflows in my repo i would need some github action to proceed furthur. it'll help me if you could suggest any.

**Reply by svrnm** [reply]
there is none that I am aware of, but the ecosystem is large, so I might just miss it, check with the CI/CD community (link above, there is a slack channel), what they know about that topic, that's your best chance.

**Reply by Samyu09** [reply]
I have two questions to ask may be you can help .

1. Does Opentelemtry provide any option to convert traces to metrics which prometheus can read at backend. Since i found this action that can generate traces of workflow.      **inception-health/otel-export-trace-action@latest**
if yes please point me to the links 

2. O regards to this action.  **inception-health/otel-export-trace-action@latest**  if the traces are sent to the opentelemetry where can i view the sent traces at the opentelemetry end.

@svrnm
